### PR TITLE
Map Screen: Show user location and button get user location

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 73909: Handle event](https://edu-redmine.sun-asterisk.vn/issues/73909)
## WHAT
- Show user location
- Action: Returns to user position when button is tapped
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HCM_iOS_2024_Weather_App/assets/161679147/47f1448f-3d6c-4b91-b7fc-8a1cfea98e46

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 1]
```diff
@@ -20,11 +21,22 @@ final class MapViewController: UIViewController {
     
     private let searchController = UISearchController()
     private let cornerRadius = 15.0
+    private let locationManager: CLLocationManager = CLLocationManager()
```
Comment (dongtg-1485) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: tách riêng việc quản lý location ra một service khác, không code logic trực tiếp vào view như thế này

**Review by dongtg-1485 — CHANGES_REQUESTED** [review]

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 2]
```diff
@@ -19,23 +19,55 @@ final class MapViewController: UIViewController {
...
+    }
+    
+    private func centerMapOnUserLocation() {
+        if let userLocation = locationManager.getCurrentLocation() {
+            let region = MKCoordinateRegion(center: userLocation.coordinate, latitudinalMeters: Constants.latitudinalMeters, longitudinalMeters: Constants.longitudinalMeters)
+            mapView.setRegion(region, animated: true)
+        }
+    }
+    
+    @IBAction func getUserLocationButtonTapped(_ sender: Any) {
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: private

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 3]
```diff
@@ -19,23 +19,55 @@ final class MapViewController: UIViewController {
...
     }
+    
+    private func configureMap() {
+        mapView.showsUserLocation = true
+        centerMapOnUserLocation()
+    }
+    
+    private func centerMapOnUserLocation() {
+        if let userLocation = locationManager.getCurrentLocation() {
+            let region = MKCoordinateRegion(center: userLocation.coordinate, latitudinalMeters: Constants.latitudinalMeters, longitudinalMeters: Constants.longitudinalMeters)
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]: này dài quá break center, latitudinalMeters,  longitudinalMeters  xuống dòng bớt e

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 4]
```diff
@@ -19,23 +19,55 @@ final class MapViewController: UIViewController {
...
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        centerMapOnUserLocation()
     }
 }
 
 extension MapViewController {
     func setUpUI() {
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift:40]: private

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift [hunk 5]
```diff
@@ -19,23 +19,55 @@ final class MapViewController: UIViewController {
     @IBOutlet weak var statusImageView: UIImageView!
     
     private let searchController = UISearchController()
-    private let cornerRadius = 15.0
+    private var getLatitude = 0.0
+    private var getLongitude = 0.0
```
Comment (ghost) [inline Weather_App/Weather_App/Scenes/Map/MapViewController.swift]:
```suggestion
    private var location = (latitude: 0.0, longitude: 0.0)
```

**Review by dongtg-1485 — APPROVED** [review]

File: Weather_App/Weather_App/Scenes/Map/MapViewController.swift (hunk 4 above)
Comment (ghost) [reply Weather_App/Weather_App/Scenes/Map/MapViewController.swift:40]: Dạ là mình thêm private trong hàm này luôn hả anh

Comment (ghost) [reply Weather_App/Weather_App/Scenes/Map/MapViewController.swift:40]: @anhb-3436 đúng r e, private những func mà dùng trong vc này

Comment (ghost) [reply Weather_App/Weather_App/Scenes/Map/MapViewController.swift:40]: Dạ em dùng internal được không anh

**Review by daolq-2712 — APPROVED** [review]
OK!

**Review by ghost — APPROVED** [review]

**Pull request merged**